### DISCUSSION TITLE
Improving the documentation on blockdata

### DISCUSSION THREAD

**Discussion by dpc** [rust-bitcoin/rust-bitcoin#1017] [Ideas]
I would like to add more docs to block-data types and structures, so e.g. someone goes to https://docs.rs/bitcoin/latest/bitcoin/blockdata/block/struct.BlockHeader.html , they can get more details about meaning and use of each field. 

The sections I'd like to include (when relevant) are:

* "Bitcoin Core references" - permalinks to Bitcoin Core implementation that correspond to the given value/type.
* "BIPs" - relevant BIPs that describe meaning of a field

Hopefully, this would make https://docs.rs/bitcoin a great educational material.

**Please let me know if you find it useful and worthwhile.** The only downside that I could see is that there will be more text (docstrings) in the code.

As a showcase, I wanted to do `BlockHeader::version` since I've happened to read through a bip about it. 

~~Current documentation states "The protocol version. Should always be 1.". However, I'm sure that `version` was used to soft-fork signaling and https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki seems still in state `final`. So is the current documentation incorrect or am I confused?~~

I guess BIP9 replaces BIP34, and existing documentation must have been written between that period?

**Comment by tcharding** [comment]
I like the idea of rust-bitcoin being a place one can learn about Bitcoin instead of having to learn Bitcoin first then hack on rust-bitcoin.

**Comment by apoelstra** [comment]
I agree. I already tend to use the source code of rust-bitcoin rather than that of Core to check on things.
